### ISSUE TITLE
Unexpected non-function AJV validator type

### ISSUE THREAD

**Issue by adrianaferrugento** [serverless/serverless#10679]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

Serverless deployment is failing occasionally, this has been happening since updating to version 3.

### Service configuration (serverless.yml) content

```yaml
service: service-name
projectDir: ../../

plugins:
  - serverless-s3-sync

custom:
  bucketName: name-${env:STAGE}
  s3Sync:
    - bucketName: ${self:custom.bucketName}
      bucketPrefix: ${env:PR_NUMBER}
      localDir: dist

provider:
  name: aws
  runtime: nodejs12.x
  stage: ${env:STAGE}

resources:
  Resources:
    StaticSite:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: ${self:custom.bucketName}
        CorsConfiguration:
          CorsRules:
            - AllowedHeaders:
                - '*'
              AllowedMethods:
                - GET
              AllowedOrigins:
                - 'https://*.${env:DOMAIN}'

    StaticSiteS3BucketPolicy:
      Type: AWS::S3::BucketPolicy
      Properties:
        Bucket:
          Ref: StaticSite
        PolicyDocument:
          Statement:
            - Effect: Allow
              Action:
                - s3:GetObject
              Resource: !Sub '${StaticSite.Arn}/*'
              Principal:
                CanonicalUser: ${cf:apps-${env:STAGE}.S3CanonicalUserId}
```

### Command name and used flags

sls deploy --verbose

### Command output

```shell
× Unexpected validator {}, resolved from source 
Environment: linux, node 16.13.2, framework 3.2.0 (local), plugin 6.0.0, SDK 4.3.1
Credentials: Local, environment variables
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Error: Unexpected non-function AJV validator type. Please report at https://github.com/serverless/serverless including all the logs output
    at getValidate (/home/runner/work/apps/apps/node_modules/.pnpm/serverless@3.2.0/node_modules/serverless/lib/classes/config-schema-handler/resolve-ajv-validate.js:73:11)
    at async ConfigSchemaHandler.validateConfig (/home/runner/work/apps/apps/node_modules/.pnpm/serverless@3.2.0/node_modules/serverless/lib/classes/config-schema-handler/index.js:98:22)
    at async Service.validate (/home/runner/work/apps/apps/node_modules/.pnpm/serverless@3.2.0/node_modules/serverless/lib/classes/service.js:267:5)
    at async Serverless.run (/home/runner/work/apps/apps/node_modules/.pnpm/serverless@3.2.0/node_modules/serverless/lib/serverless.js:162:28)
    at async /home/runner/work/apps/apps/node_modules/.pnpm/serverless@3.2.0/node_modules/serverless/scripts/serverless.js:686:9
undefined
```

### Environment information

```shell
Framework Core: 3.2.0 (local)
Plugin: 6.0.0
SDK: 4.3.1
```

**Comment by medikoo** [comment]
> It's happened several times but usually after rerunning it will work

Ok, so you've observed it on few occasions, but in all cases the second run (one made directly after when error occurred) worked

**Comment by adrianaferrugento** [comment]
Yes exactly, just to add that I'm seeing this when I run it with my Github Actions workflow, so it's not on my machine. So rerunning the workflow makes it work pretty much all the time.

**Comment by adrianaferrugento** [comment]
I've updated to the latest version but getting a new error now, happy to open a separate issue with this 🙂 

```
Waiting for new change set to be created

× Stack apps-navigation-shell-dev failed to deploy (2s)
Environment: linux, node 16.13.2, framework 3.2.1 (local), plugin 6.0.0, SDK 4.3.1
Credentials: Local, environment variables
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:

ChangeSet [apps-navigation-shell-dev-change-set] does not exist
1 deprecation found: run 'serverless doctor' for more details
Waiting for new change set to be created
```

**Comment by pgrzesik** [comment]
Sorry you've run into trouble @adrianaferrugento - could you please open a separate task for this? It seems to be an unrelated error, thanks in advance 🙇

**Comment by pgrzesik** [comment]
I'm going to close this one as it's supposed to be solved with 3.2.1
